### ISSUE TITLE
New Rule: Duplicate Property

### ISSUE THREAD

**Issue by Snugug** [sasstools/sass-lint#28]
As Per [SCSS Lint](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#duplicateproperty)

**Comment by bgriffith** [comment]
Will take a look at this one today.

**Comment by bgriffith** [comment]
Any opinions on whether this should trigger when using both long hand and short hand properties?

Should the following trigger an error?
```scss
.foo {
  margin-left: 10px;
  padding: 0;
  margin: 0;
}
```

**Comment by DanPurdy** [comment]
I reckon it should because you are technically duplicating it there it's just sort of hidden from you

```
margin: {
  margin-top: 0,
  margin-right: 0,
  margin-bottom: 0,
  margin-left: 0
}
```

**Comment by bgriffith** [comment]
Think the best action would be to add it as an option that is set to true by default, but allows the user to ignore. What do you reckon?

**Comment by bgriffith** [comment]
I've checked out the way SCSS Lint works, and they don't cause an error as technically they are different properties. Maybe this could be a different rule.

**Comment by DanPurdy** [comment]
Makes sense to me, maybe that would actaully be preferable so we can keep the rules fairly granular.

**Comment by bgriffith** [comment]
Since this is only ever disallowing (as you wouldn't ever be enforcing duplicate properties, I think it should follow the current convention for 'disallows' and call it `no-duplicate-property`
